feat(minifier): compress `Object(expr)(args)` to `(0, expr)(args)`

sapphi-red opened the pull request:
Compress `Object(expr)(args)` to `(0, expr)(args)`.

This can be done because
- the function call throws an error if the callee is not a function.
- `Object(expr)` would output a function if and only if `expr` is a function.

---

**References**

- [spec of `Object()`](https://tc39.es/ecma262/2025/multipage/fundamental-objects.html#sec-object-value)

magic-akari commented:
I'm wondering where the `Object(expr)(args)` pattern originates from? Is it typically generated by compilers like Babel, or does it come from another source?

sapphi-red committed:
feat(minifier): compress `Object(expr)(args)` to `(0, expr)(args)` (#13092)

Compress `Object(expr)(args)` to `(0, expr)(args)`.

This can be done because
- the function call throws an error if the callee is not a function.
- `Object(expr)` would output a function if and only if `expr` is a function.

---

**References**

- [spec of `Object()`](https://tc39.es/ecma262/2025/multipage/fundamental-objects.html#sec-object-value)
